### ISSUE TITLE
Problems running on MacOS 10.12.6

### ISSUE THREAD

**Issue by a-leithner** [plk/biber#325]
Within a project of mine, I provide users with the ability to create custom "compilation workflows" for LaTeX documents, including Biber runs. However, running the latest Biber version from CTAN on a MacOS 10.12.6 device, I get the following errors which seem to be somewhat related to #187. I have installed TeX Live 2020 using the provided Unix installer (I'm not using MacTeX from my program because it acts as an editor and thus all extra stuff shipped with MacTeX is not necessary) and am running programs from within the `x86_64-darwinlegacy` binary folder. The full `stderr` output (sensitive paths stripped) is as follows (extra output such as the first and last two lines are from my program):

```
[#2] Executing `/Users/user/texlive/2020/bin/x86_64-darwinlegacy/biber somefile'...
Can't load '/var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/b322188f.bundle' for module XML::LibXML: dlopen(/var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/b322188f.bundle, 1): Library not loaded: /opt/local/lib/libicui18n.58.dylib
  Referenced from: /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T//par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/libxml2.2.dylib
  Reason: image not found at /loader/HASH(0x10101c838)/DynaLoader.pm line 197.
 at /loader/HASH(0x101074dd8)/PAR/Heavy.pm line 140.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/XML/LibXML.pm line 156.
Compilation failed in require at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/XML/LibXML/Simple.pm line 17.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/XML/LibXML/Simple.pm line 17.
Compilation failed in require at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/LaTeX/Recode.pm line 16.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/LaTeX/Recode.pm line 16.
Compilation failed in require at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/Utils.pm line 19.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/Utils.pm line 19.
Compilation failed in require at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/Internals.pm line 8.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber/Internals.pm line 8.
Compilation failed in require at /loader/HASH(0x1010b6700)/parent.pm line 16.
BEGIN failed--compilation aborted at /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-6d69616d61796572/cache-487b97104ab64f45b54915cf8abf61a5c48774e9/inc/lib/Biber.pm line 5.
Compilation failed in require at script/biber-darwin line 17.
BEGIN failed--compilation aborted at script/biber-darwin line 17.

Process exited with code 2
Execution finished
```

Since I'm not used working with MacOS, I don't know where to start troubleshooting this issue and am thus seeking help here. As a side note, this very same version of Biber was working on this very system before, and no OS updates have been applied in between runs, according to the user. What I have tried so far was force-reinstalling Biber and BibLaTeX via `tlmgr` and updating both, of which none produced any different result.

If you need more information, please let me know and I will reach out to my end-user, although this might take some time.

**Comment by plk** [comment]
First thing is to clear the cache - can you delete:

```
/var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-*
```

You will likely have to do this in a terminal window:

```
\rm -rf /var/folders/wt/xk7v2grj4s56z9dghsfsc9gc0000gn/T/par-*
```

**Comment by a-leithner** [comment]
Thank you for this very quick response! I will reach out to my end-user and once the cache has been cleared, I will come back to this issue given the problem persists.

**Comment by moewew** [comment]
The error message reminds me of https://github.com/plk/biber/issues/298

**Comment by plk** [comment]
Well spotted - I think this is almost certainly the issue. The caveat is now that you will almost certainly need `biblatex` 3.15 to use the 2.15 DEV version of `biber` now. Which leads me to think, @moewew, that we should probably release 3.15/2.15?

**Comment by moewew** [comment]
@plk  Yes, it is probably time for another release. There are 5 issues in the 'in progress' column at https://github.com/plk/biblatex/projects/9 and three open pull requests that I'd like to handle before we release.

* I'd probably defer the `@patent` date PR https://github.com/plk/biblatex/pull/907 for another day, but at the moment I lean towards not merging it.
* I'd appreciate a comment on the names in https://github.com/plk/biblatex/pull/1031 (and I should probably add a short note to the change log when that is merged).
* In general I'm not too sure about the idea in https://github.com/plk/biblatex/pull/1028. I liked it very much initially since it gave me a way out of the 'commands in bibstring' dilemma. But now that I left it sitting there for a bit, I'd like to hear a comment on whether that is too *ad hoc*.

In most of the issues in the 'in progress' column I commented something but either didn't hear back from the original poster or wasn't too sure how to proceed (when I was the reporter).

**Comment by a-leithner** [comment]
@plk, as I already had anticipated from this very helpful reply by @moewew, clearing these caches sadly did not work as expected. Is there any ETA for the fixed release? Btw, thank you all for this fine piece of software and quick support!

**Comment by moewew** [comment]
@plk Given our discussions, I'd say we release the current `dev` and don't merge https://github.com/plk/biblatex/pull/907, https://github.com/plk/biblatex/pull/1031 and https://github.com/plk/biblatex/pull/1028 for the time being. Enough changes have built up over time and fixing this Biber issue is quite a good reason to go for a release now. We can then think about how to handle these three PRs and a few other open issues after releasing 3.15.

**Comment by plk** [comment]
I'll start the release process for 2.15/3.15

**Comment by moewew** [comment]
@plk Great! I noticed that the title page of the docs in `biblatex-3.15.tgz` on SF still says "3.14". Maybe something went wrong when the archives were packaged. The title page of the docs in the TDS archive looked fine.

I'll mention the new release in #842 and on c.t.t.

**Comment by plk** [comment]
Strange how this sometimes happens - fixed now (with exactly the same process which caused the problem ...). I think SF replication sometimes fails/breaks.

**Comment by a-leithner** [comment]
I can now confirm this issue resolved with Biber 2.15. Thank you all for such a rapid fix!
